client/src/components/RegisterUserModal.jsx

```diff
@@ -0,0 +1,166 @@
+import {
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Input,
+  Select,
+  useToast,
+} from '@chakra-ui/react';
+import { StyledButton } from './views/Login.jsx';
+import { Text } from './common/Typography.jsx';
+import React, { Fragment, useRef, useState } from 'react';
+import styled from '@emotion/styled';
+import { makeRequest } from '../utils/api.js';
+
+const ValidationText = styled(Text)`
+  font-size: 10px;
+  padding-left: 12px;
+`;
+
+const rolesAvailable = ['Admin', 'General'];
```

Comment:
I would just move this to a constants file for now just so we can use it later on for the next PRs. All good from me after this

Reply:
Very sound advice